infoblox/resource_infoblox_a_record.go

```diff
@@ -201,6 +212,31 @@ func resourceARecordGet(d *schema.ResourceData, m interface{}) error {
 }
 
 func resourceARecordUpdate(d *schema.ResourceData, m interface{}) error {
+	var updateSuccessful bool
+	defer func() {
+		// Reverting the state back, in case of a failure,
+		// otherwise Terraform will keep the values, which leaded to the failure,
+		// in the state file.
+		if !updateSuccessful {
+			prevNetView, _ := d.GetChange("network_view")
+			prevDNSView, _ := d.GetChange("dns_view")
+			prevFQDN, _ := d.GetChange("fqdn")
+			prevIPAddr, _ := d.GetChange("ip_addr")
+			prevCIDR, _ := d.GetChange("cidr")
+			prevTTL, _ := d.GetChange("ttl")
+			prevComment, _ := d.GetChange("comment")
+			prevEa, _ := d.GetChange("ext_attrs")
+
+			_ = d.Set("network_view", prevNetView.(string))
```

Comment:
This place is special: we already return an error, we can catch one more error (from d.Set()) but we cannot return it, because the main error message will be lost. And the code is still not utilizing new Terraform's framework which allows to return multiple errors/warnings.

We can log a message, but let's discuss this.